### PR TITLE
Rewrite file-access using NSURL-API as per Apple's reccommendation

### PR DESCRIPTION
**Pull request by MatkovIvan**
Replace NSString for paths everywhere and use NSURL instead.

### CODE REVIEW

File: MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m [hunk 1]
```diff
@@ -165,38 +165,36 @@ - (void)renewCurrentFileForStorageKey:(NSString *)storageKey {
...
-      _baseDirectoryPath = [appSupportPath stringByAppendingPathComponent:kMSLogsDirectory];
+- (NSURL *)baseDirectoryURL {
+  if (!_baseDirectoryURL) {
+    NSURL *appSupportURL = [[[NSFileManager defaultManager] URLsForDirectory:NSApplicationSupportDirectory inDomains:NSUserDomainMask] lastObject];
+    if (appSupportURL) {
+      _baseDirectoryURL = (NSURL * _Nonnull)[appSupportURL URLByAppendingPathComponent:kMSLogsDirectory];
     }
 
-    MSLogVerbose([MSMobileCenter logTag], @"Storage Path:\n%@", _baseDirectoryPath);
+    MSLogVerbose([MSMobileCenter logTag], @"Storage Path:\n%@", self->_baseDirectoryURL);
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m]: Is there a reason to not call _baseDirectoryURL directly here.

File: MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m [hunk 2]
```diff
@@ -43,87 +43,86 @@ - (NSFileManager *)fileManager {
...
+  if (![directoryURL checkResourceIsReachableAndReturnError:&error]) {
     return nil;
   }
 
   NSMutableArray<MSFile *> *files;
-  NSError *error;
-  NSArray *allFiles = [fileManager contentsOfDirectoryAtPath:path error:&error];
+  NSArray *allFiles = [fileManager contentsOfDirectoryAtURL:(NSURL * _Nonnull)directoryURL
+                                 includingPropertiesForKeys:@[NSURLCreationDateKey]
+                                                    options:NSDirectoryEnumerationSkipsHiddenFiles
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: I would not skip hidden files to be more future proof.
@jaelim-ms @TroubleMakerBen Any though?

File: MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m (hunk 1 above)
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m]: Can this be done without direct ivar access?

File: MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m (hunk 2 above)
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: I agree that we shouldn't skip hidden files.

Comment (MatkovIvan) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: [link](https://developer.apple.com/reference/foundation/nsfilemanager/1413768-contentsofdirectoryaturl) - "the only supported option is `NSDirectory​Enumeration​Skips​Hidden​Files`"

Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: I think what the others say is to pass `nil`. @MatkovIvan. Doesn't it work?

File: MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m (hunk 1 above)
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m]: why not `self.baseDirectoryURL`?

File: MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m (hunk 2 above)
Comment (TroubleMakerBen) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: I didn't look at the API, if `NSDirectory​Enumeration​Skips​Hidden​Files` this is fine. We don't create any hidden files anyway.

File: MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m (hunk 1 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m]: Yep but it's not the only var accessed directly, if we go that way we should propertify them all in this class.

Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m]: Yep should be a property you are right.

File: MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m (hunk 2 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileUtil.m]: At the end if the options can't be `nil` then yes we can use that option.

File: MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m (hunk 1 above)
Comment (MatkovIvan) [inline MobileCenter/MobileCenter/Internals/Storage/MSFileStorage.m]: This is auto-fix by XCode for fixing warning:
`Block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior`
Also with `self.baseDirectoryURL` this is recursion.

**Comment by MatkovIvan** [comment]
@clpolet please let me know if there is anything which should be updated.

**Pull request merged**